src/sdk/namespace_fs.js

```diff
@@ -1302,6 +1362,49 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * Return false if the entry is outside of the bucket
+     * @param {*} fs_account_config 
+     * @param {*} entry_path 
+     * @returns 
+     */
+    async _is_path_in_bucket_boundaries(fs_account_config, entry_path) {
+        dbg.log0('check_bucket_boundaries: fs_account_config', fs_account_config, 'file_path', entry_path);
+        if (!entry_path.startsWith(this.bucket_path)) {
+            dbg.log0('check_bucket_boundaries: the path', entry_path, 'is not in the bucket', this.bucket_path, 'boundaries');
+                return false;
+        }
+        try {
+            // Returns tyhe real path of the entry.
+            // The entry path may point to regular file or directory, but can have symbolic links  
+            let full_path = await nb_native().fs.realpath(fs_account_config, entry_path);
+            if (!full_path.startsWith(this.bucket_path)) {
+                dbg.log0('check_bucket_boundaries: the path', entry_path, 'is not in the bucket', this.bucket_path, 'boundaries');
+                return false;
+            }
+        } catch (err) {
+            // Error: No such file or directory
+            // In the upload use case, the destination file desn't exist yet, need to validate the parent dirs path.
+            if (err.code === 'ENOENT') {
+                return this._is_path_in_bucket_boundaries(fs_account_config, path.dirname(entry_path));
+            }
+            dbg.log0('check_bucket_boundaries error', err.code, entry_path, err);
+            return false;
```

Comment:
Do we want to hide the error? it seems like in some cases we will return EACCESS when the fs error is different.

Reply:
The error log is for unusual errors, like "I/O", "Too many symbolic links" and so on.
So, I think we should know about that.

Reply:
Just asking if the user shouldn't be aware of those as you only log them but not sending them back to the user - he will get just access denied. Again, just a question - not sure if that's good or not...

Reply:
I think no, but @dannyzaken wdyt?

Reply:
Done